lively.merger/merger.js

```diff
@@ -1,107 +1,121 @@
+import MorphicDB from 'lively.morphic/morphicdb/db.js';
 
-export class Merger {
-  static propertiesFromMorph (morph) {    
-    const properties = {};
+export function propertiesFromMorph (morph) {    
+  const properties = {};
 
-    Object.filter = (obj, predicate) => 
-      Object.keys(obj)
-        .filter(key => predicate(obj[key]))
-        .reduce((res, key) => (res[key] = obj[key], res), {});
-    
-    Object.keys(Object.filter(morph.propertiesAndPropertySettings().properties, property => !property.derived)).forEach(key => {
-      if (key !== ('styleProperties' || 'style')) { properties[key] = morph[key]; }
-    });
-    return properties;
-  }
+  Object.filter = (obj, predicate) => 
+    Object.keys(obj)
+      .filter(key => predicate(obj[key]))
+      .reduce((res, key) => (res[key] = obj[key], res), {});
+  Object.keys(Object.filter(morph.propertiesAndPropertySettings().properties, property => !property.derived)).forEach(key => {
+    if (key !== ('styleProperties' || 'style')) { properties[key] = morph[key]; }
+  });
+  return properties;
+}
 
-  static mergeMorphsWithIdsIntoA (morphAid, morphBid) {
-    return this.mergeMorphsWithIds(morphAid, morphBid, (properties, mergeConflicts) => {
-      const morphA = $world.submorphs.filter(morph => morph.id === morphAid)[0];
-      Object.keys(properties).forEach(key => {
-        morphA[key] = properties[key];
-      });
-      return morphA;
-    });
+export async function getLowestCommonAncestor (morphA, morphB) {
+  for (let index = morphA.derivationIds.length - 1; index >= 0; index--) {
+    const currentId = morphA.derivationIds[index];
+    if (morphB.derivationIds.includes(currentId)) {
+      const parentMorph = $world.submorphs.filter(morph => morph.id === currentId)[0];
+      if (parentMorph) {
+        return parentMorph;
+      } else {
+        const latestCommit = await MorphicDB.default.fetchCommit('world', $world.name);
+        const chronologicalHistory = await MorphicDB.default.log(latestCommit._id, undefined, true);
+        let lowestCommit;
+        for (const commit of chronologicalHistory) {
+          const worldStateForCommit = await MorphicDB.default.fetchCommit('world', $world.name, commit._id);
+          const snapshot = await MorphicDB.default.fetchSnapshot(undefined, undefined, worldStateForCommit._id);
+          const submorphs = snapshot.snapshot[Object.keys(snapshot.snapshot).filter(name => name.startsWith('livelyworld_'))].props.submorphs.value;
+          if (submorphs.filter(submorph => submorph.id === currentId).length > 0) {
+            lowestCommit = commit;
+            break;
+          }
+        }
+        const snapshot = await MorphicDB.default.fetchSnapshot(undefined, undefined, lowestCommit._id);
+        const world = await loadMorphFromSnapshot(snapshot);
+        const morph = world.submorphs.filter(submorph => submorph.derivationIds[submorph.derivationIds.length - 2] === currentId)[0];
+        return morph;
```

Comment:
Just to make it a bit more concise

```suggestion
        return world.submorphs.filter(submorph => submorph.derivationIds[submorph.derivationIds.length - 2] === currentId)[0];
```